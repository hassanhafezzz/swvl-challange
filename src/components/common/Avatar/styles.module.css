@value utils: '../../../styles/utils.css';

.media {
  padding: 20px;
  display: flex;
  align-items: center;
  justify-content: flex-start;
  background-color: var(--color-gray-100);
  border-radius: 5px;
}

.media.missed,
.media.cancelled {
  background-color: var(--color-bg-red);
}

.media.completed {
  background-color: var(--color-bg-green);
}

.image {
  width: 80px;
  height: 80px;
  border-radius: 50%;
}

.body {
  display: flex;
  flex-direction: column;
  margin-left: 15px;
}

.body p {
  margin: 0;
}

.body p + p {
  margin-top: 8px;
}

.title {
  font-size: var(--h5-size);
  font-weight: var(--bold);
}
