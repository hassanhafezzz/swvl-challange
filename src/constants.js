// booking
export const FILL_ALL_BOOKINGS = 'FILL_ALL_BOOKINGS';
export const ADD_BOOKING = 'ADD_BOOKING';
export const UPDATE_BOOKER_STATUS = 'UPDATE_BOOKER_STATUS';

// trip
export const RESET_TRIP = 'RESET_TRIP';
export const START_TRIP = 'START_TRIP';
export const END_TRIP = 'END_TRIP';

// location
export const SET_STATIONS = 'SET_STATIONS';
export const SET_DIRECTION = 'SET_DIRECTION';
export const UPDATE_CURRENT_DISTANCE = 'UPDATE_CURRENT_DISTANCE';
export const UPDATE_STATIONS_DISTANCE_AND_ETA =
  'UPDATE_STATIONS_DISTANCE_AND_ETA';
export const UPDATE_STATION_ARRIVAL_STATUS = 'UPDATE_STATION_ARRIVAL_STATUS';

// status
export const TRIP_NOT_STARTED = 'TRIP_NOT_STARTED';
export const TRIP_IN_PROGRESS = 'TRIP_IN_PROGRESS';
export const TRIP_COMPLETED = 'TRIP_COMPLETED';

export const LATENCY = {
  EARLY: 'early',
  ON_TIME: 'on time',
  LATE: 'late',
};

export const PAYMENT_METHODS = {
  CASH: 'cash',
  CREDIT: 'credit',
};

export const BOOKING_STATUS = {
  BOOKED: 'booked',
  COMPLETED: 'completed',
  MISSED: 'missed',
  CANCELLED: 'cancelled',
};
