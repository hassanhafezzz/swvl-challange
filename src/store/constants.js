export const INITIALIZE_APP = 'INITIALIZE_APP';
// booking
export const FILL_ALL_BOOKINGS = 'FILL_ALL_BOOKINGS';
export const ADD_BOOKING = 'ADD_BOOKING';
export const UPDATE_BOOKER_STATUS = 'UPDATE_BOOKER_STATUS';

// trip
export const RESET_TRIP = 'RESET_TRIP';
export const START_TRIP = 'START_TRIP';
export const END_TRIP = 'END_TRIP';

// location
export const UPDATE_ROUTE = 'UPDATE_ROUTE';
export const UPDATE_CURRENT_DISTANCE = 'UPDATE_CURRENT_DISTANCE';
export const UPDATE_STATION_INFO = 'UPDATE_STATION_INFO';

// status
export const TRIP_NOT_STARTED = 'TRIP_NOT_STARTED';
export const TRIP_IN_PROGRESS = 'TRIP_IN_PROGRESS';
export const TRIP_COMPLETED = 'TRIP_COMPLETED';

export const LATENCY_EARLY = 'LATENCY_EARLY';
export const LATENCY_ON_TIME = 'LATENCY_ON_TIME';
export const LATENCY_LATE = 'LATENCY_LATE';
