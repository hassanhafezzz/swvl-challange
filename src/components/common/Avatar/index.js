import React from 'react';
import classNames from 'classnames/bind';
import styles from './styles.module.css';
import { BOOKING_STATUS } from '../../../constants';

const cx = classNames.bind(styles);

export default function Avatar({
  img,
  name,
  status,
  tripsCount,
  pickupStation,
  dropOffStation,
}) {
  return (
    <div
      data-testid="user"
      className={cx('media', {
        booked: status === BOOKING_STATUS.BOOKED,
        missed: status === BOOKING_STATUS.MISSED,
        cancelled: status === BOOKING_STATUS.CANCELLED,
        completed: status === BOOKING_STATUS.COMPLETED,
      })}
    >
      <img className={cx('image')} src={img} alt={name} loading="lazy" />
      <div className={cx('body')}>
        <p className={cx('title')}>{name}</p>
        <p>
          <b>Status</b> {status}
        </p>
        <p>
          <b>Trips</b> {tripsCount}
        </p>
        <p>
          <b>From</b> {pickupStation}
        </p>
        <p>
          <b>To</b> {dropOffStation}
        </p>
      </div>
    </div>
  );
}
