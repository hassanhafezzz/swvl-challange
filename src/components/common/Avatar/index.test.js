import React from 'react';
import { render } from '@testing-library/react';
import { BOOKING_STATUS } from '../../../constants';
import Avatar from './index';

const userMock = {
  id: 'e5da1a8f9b844d56a4a32cbaa590254e',
  name: 'Ahmed Ossama',
  tripsCount: 210,
  image:
    'https://scontent.fcai19-1.fna.fbcdn.net/v/t31.0-8/20424255_10159017225385401_82881391294765437_o.jpg?_nc_cat=105&_nc_sid=09cbfe&_nc_ohc=2KLOtnblWN0AX_7STbA&_nc_ht=scontent.fcai19-1.fna&oh=6b0200cf4b3f857ef5d363a4f119b073&oe=5EF3C80B',
  pickupStation: 'almaza',
  dropOffStation: 'city stars',
  paymentMethod: 'cash',
  status: BOOKING_STATUS.BOOKED,
};

test('it should display user avatar', () => {
  const { getByTestId, getByAltText } = render(<Avatar {...userMock} />);

  const user = getByTestId('user');
  expect(user).toHaveTextContent('Ahmed Ossama');
  expect(user).toHaveTextContent(BOOKING_STATUS.BOOKED);
  expect(user).toHaveTextContent(210);
  expect(user).toHaveTextContent('almaza');
  expect(user).toHaveTextContent('city stars');
  expect(getByAltText('Ahmed Ossama')).toBeTruthy();
});

test('add snapshot for user avatar', () => {
  const { container, asFragment } = render(<Avatar {...userMock} />);
  expect(asFragment(container)).toMatchSnapshot();
});
