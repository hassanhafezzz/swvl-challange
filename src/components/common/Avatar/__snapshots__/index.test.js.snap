// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`add snapshot for user avatar 1`] = `
<DocumentFragment>
  <div
    class="media booked"
    data-testid="user"
  >
    <img
      alt="Ahmed Ossama"
      class="image"
    />
    <div
      class="body"
    >
      <p
        class="title"
      >
        Ahmed Ossama
      </p>
      <p>
        <b>
          Status
        </b>
         booked
      </p>
      <p>
        <b>
          Trips
        </b>
         210
      </p>
      <p>
        <b>
          From
        </b>
         almaza
      </p>
      <p>
        <b>
          To
        </b>
         city stars
      </p>
    </div>
  </div>
</DocumentFragment>
`;
